websocketOutportDriver/websocketOutportDriverNodePart.go

```diff
@@ -0,0 +1,202 @@
+	err := o.handleAction(args, outportSenderData.OperationSaveBlock)
+	if err != nil {
+		return fmt.Errorf("%w on SaveBlock", err)
+	}
+
+	return nil
+}
+
+// RevertIndexedBlock will handle the action of reverting the indexed block
+func (o *websocketOutportDriverNodePart) RevertIndexedBlock(header data.HeaderHandler, body data.BodyHandler) error {
+	args := outportSenderData.ArgsRevertIndexedBlock{
+		Header: header,
+		Body:   body,
+	}
+	err := o.handleAction(args, outportSenderData.OperationRevertIndexedBlock)
+	if err != nil {
+		return fmt.Errorf("%w on RevertIndexedBlock", err)
+	}
+
+	return nil
+}
+
+// SaveRoundsInfo will handle the saving of rounds
+func (o *websocketOutportDriverNodePart) SaveRoundsInfo(roundsInfos []*indexer.RoundInfo) error {
+	args := outportSenderData.ArgsSaveRoundsInfo{
+		RoundsInfos: roundsInfos,
+	}
+	err := o.handleAction(args, outportSenderData.OperationSaveRoundsInfo)
+	if err != nil {
+		return fmt.Errorf("%w on SaveRoundsInfo", err)
+	}
+
+	return nil
+}
+
+// SaveValidatorsPubKeys will handle the saving of the validators' public keys
+func (o *websocketOutportDriverNodePart) SaveValidatorsPubKeys(validatorsPubKeys map[uint32][][]byte, epoch uint32) error {
+	args := outportSenderData.ArgsSaveValidatorsPubKeys{
+		ValidatorsPubKeys: validatorsPubKeys,
+		Epoch:             epoch,
+	}
+	err := o.handleAction(args, outportSenderData.OperationSaveValidatorsPubKeys)
+	if err != nil {
+		return fmt.Errorf("%w on SaveValidatorsPubKeys", err)
+	}
+
+	return nil
+}
+
+// SaveValidatorRating will handle the saving of the validators' rating
+func (o *websocketOutportDriverNodePart) SaveValidatorsRating(indexID string, infoRating []*indexer.ValidatorRatingInfo) error {
+	args := outportSenderData.ArgsSaveValidatorsRating{
+		IndexID:    indexID,
+		InfoRating: infoRating,
+	}
+	err := o.handleAction(args, outportSenderData.OperationSaveValidatorsRating)
+	if err != nil {
+		return fmt.Errorf("%w on SaveValidatorsRating", err)
+	}
+
+	return nil
+}
+
+// SaveAccounts will handle the accounts' saving
+func (o *websocketOutportDriverNodePart) SaveAccounts(blockTimestamp uint64, acc []data.UserAccountHandler) error {
+	args := outportSenderData.ArgsSaveAccounts{
+		BlockTimestamp: blockTimestamp,
+		Acc:            acc,
+	}
+	err := o.handleAction(args, outportSenderData.OperationSaveAccounts)
+	if err != nil {
+		return fmt.Errorf("%w on SaveAccounts", err)
+	}
+
+	return nil
+}
+
+// FinalizedBlock will handle the finalized block
+func (o *websocketOutportDriverNodePart) FinalizedBlock(headerHash []byte) error {
+	args := outportSenderData.ArgsFinalizedBlock{
+		HeaderHash: headerHash,
+	}
+	err := o.handleAction(args, outportSenderData.OperationFinalizedBlock)
+	if err != nil {
+		return fmt.Errorf("%w on FinalizedBlock", err)
+	}
+
+	return nil
+}
+
+// Close will handle the closing of the outport driver web socket sender
+func (o *websocketOutportDriverNodePart) Close() error {
+	// TODO: close the web socket here
```

Comment:
this is important. Finish the todo? o.webSocketSender could implement the io.Closer interface

Reply:
done by using a channel